integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/JMXClient.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.esb.integration.common.utils.clients;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import javax.management.InstanceNotFoundException;
+import javax.management.ListenerNotFoundException;
+import javax.management.MBeanServerConnection;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.Hashtable;
+
+/**
+ * JMX Client which can be used to connect and invoke operations on local JMX Services
+ */
+public class JMXClient {
+
+    private MBeanServerConnection mbsc = null;
+
+    private String userName;
+    private String password;
+    private ObjectName objectName;
+    private String hostName;
+    private String rmiServerPort;
+    private String rmiRegistryPort;
+
+    private JMXConnector jmxc;
+    private static final Log log = LogFactory.getLog(JMXClient.class);
+
+    /**
+     * @param connectionName - full name of the service with the connection
```

Comment:
We don't use `-` to separate out the param description. Better to avoid this in future formatting.

Reply:
Fixed with https://github.com/wso2/product-ei/pull/2148